janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -151,6 +151,10 @@
             new ConfigOption<>(ES_CREATE_NS, "use-external-mappings",
             "Whether JanusGraph should make use of an external mapping when registering an index.", ConfigOption.Type.MASKABLE, false);
 
+    public static final ConfigOption<Boolean> USE_DYNAMIC_TEMPLATE =
+            new ConfigOption<>(ES_CREATE_NS, "use-dynamic-template",
```

Comment:
This isn't specific to the case where a user's external mapping defines dynamic templates, right? Maybe `allow-mapping-update` or something like that would be a more clear name?